### ISSUE TITLE
Gnome-shell 3.24 on Antergos -- screen flicker on playback

### ISSUE THREAD

**Issue by dtschmitz** [vkohaupt/vokoscreen#181]
Hi,
I've not used vokoscreen in a while (maybe 6 months) but today wanted to record something.  When I play the recording, it shows vokoscreen on top of the desktop area flickering and video is out of sync with audio.
I've just compiled your current github version bab2539 and find the same issue persists.

video output: mkv audio: pulseaudio linux kernel: 4.11.3 gui: gnome-shell 3.24. using Gnome X.org (vs. Wayland) I get the same error with Wayland.

Please review and advise.  Thank you for all your great work.
Kind Regards
Dietrich

**Comment by tuxmuck** [comment]
Welche Linux Distribution und welche Version?

Schicke mir bitte mal ein Testvideo an die Supportadresse. Bildschirm-Vollbild in dem auch das vokoscreen Programmfenster zu sehen ist.

**Comment by dtschmitz** [comment]
Ich laufe Antergos (Arch Derivat) Linux, Rolling Release, also ist es immer neues, kein statisches Paketmanagement wie Fedora, Ubuntu.

[vokoscreen-2017-06-09_16-07-19.zip](https://github.com/vkohaupt/vokoscreen/files/1065144/vokoscreen-2017-06-09_16-07-19.zip)

**Comment by vkohaupt** [comment]
Hi Dietrich,

i have to time three resultions, show here

http://linuxecke.volkoh.de/vokoscreen/vokoscreen.html

**Comment by vkohaupt** [comment]
http://linuxecke.volkoh.de/vokoscreen/vokoscreen-tips.html

**Comment by dtschmitz** [comment]
I am using a core i5 with integrated intel gma HD and gnome-shell 2.4.
I can go to gconf-settings and disable 3D effects.  Is that what you want done?
That wouldn't be an optimal  fix.
Thank you

**Comment by dtschmitz** [comment]
[vokoscreen-2017-06-09_18-16-42.zip](https://github.com/vkohaupt/vokoscreen/files/1065359/vokoscreen-2017-06-09_18-16-42.zip)
When I disable animations (no OpenGL) I get the above video.  Note the mouse can be seen moving but when I change tabs in Chrome, none of the canvas area repaints. :(
I don't think disable animations is the appropriate answer here.
How may I help you to solve this?

**Comment by vkohaupt** [comment]
I have the video with avidemux image for image analyzed. I think it's a 
fake. The video will always re-flash, and the vokoscreen window will 
resize as it is when minimizing the window through animation (composites).

**Comment by tuxmuck** [comment]
Ich habe eben mal Antergos mit Gnome in einer VM installiert -
Dann vokoscreen über dessen Paketverwaltung eingespielt -

Es läuft zwar etwas ruckelig, aber flackern habe ich nicht.

**Comment by tuxmuck** [comment]
In dem Flacker-Video zeigt das vokoscreen Fenster weder die frame-Rate noch die aktuelle Dateigrösse an - müsste es aber.
Wenn das Video "echt" ist - geht das gar nicht.

**Comment by debuglevel** [comment]
If the video would record correctly, the youtube video should also move. Of course, if the whole video does not record properly, the vokoscreen window itself will not be recorded properly likewise.

Instead of suspecting an user to be a phony, we could do something to discover the source the problem:

Hi @dtschmitz,
could you please attach the log file (respectively the debug output if you start vokoscreen in a terminal)? Maybe the underlying ffmpeg command already reports some warnings about the problem you are encountering.

Unfortunately, the vast number of combinations of different video cards, driver versions and Linux distributions makes it sometimes very hard to test and debug software appropriately. Therefore we're happy about every user, which makes the effort to report file a bug report instead of just using another software. Thank you!

**Comment by vkohaupt** [comment]
Marc Kohaupt schrieb am 10.06.2017 um 14:08:
> Instead of suspecting an user to be a phony, we could do something to discover the source the problem:

This is no screen flicker, by screen flicker the vokoscreen window will not change the size, but in the Video we see many resized vokoscreen windows. Here two screenshots. And on the Video we see only the vokoscreen windows change the size, all other is normal.

![screenshot_20170610_145441](https://user-images.githubusercontent.com/3152567/27002993-cffcf860-4dee-11e7-9a26-de6b1228237f.png)
![screenshot_20170610_145543](https://user-images.githubusercontent.com/3152567/27002994-d002761e-4dee-11e7-937c-c73c2bd59215.png)

**Comment by debuglevel** [comment]
In the last 17 years, I've seen Linux GUIs doing so much ridiculous nonsense, that I'm willing to threat every issue in the graphical subsystem of Linux as a legitimate bug.

**Comment by dtschmitz** [comment]
I ran from terminal: $vokoscreen > debug.log 2&1 start, did a few screen movements, stop, exit.
there is no stderr output in debug.log but here are the files for your review.

Please let me know how I can be of further assistance.  -- Dietrich
[vokoscreen-2017-06-10_10-54-53.zip](https://github.com/vkohaupt/vokoscreen/files/1065893/vokoscreen-2017-06-10_10-54-53.zip)

[ffmpeg-20170610-105453.zip](https://github.com/vkohaupt/vokoscreen/files/1065894/ffmpeg-20170610-105453.zip)
[debug.zip](https://github.com/vkohaupt/vokoscreen/files/1065895/debug.zip)

**Comment by dtschmitz** [comment]
Just curious: what do you support besides ffmpeg since it is creating the flickering output. ?

[dts@ThinkPad vokoscreen]$ ffmpeg -version
ffmpeg version 3.3.1 Copyright (c) 2000-2017 the FFmpeg developers
built with gcc 7.1.1 (GCC) 20170516
configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-avisynth --enable-avresample --enable-fontconfig --enable-gmp --enable-gnutls --enable-gpl --enable-ladspa --enable-libass --enable-libbluray --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libiec61883 --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxvid --enable-shared --enable-version3
libavutil      55. 58.100 / 55. 58.100
libavcodec     57. 89.100 / 57. 89.100
libavformat    57. 71.100 / 57. 71.100
libavdevice    57.  6.100 / 57.  6.100
libavfilter     6. 82.100 /  6. 82.100
libavresample   3.  5.  0 /  3.  5.  0
libswscale      4.  6.100 /  4.  6.100
libswresample   2.  7.100 /  2.  7.100
libpostproc    54.  5.100 / 54.  5.100
[dts@ThinkPad vokoscreen]$

**Comment by debuglevel** [comment]
We provide a statically linked version of vokoscreen, which is well tested (at least on some development machines) and includes its own version of ffmpeg: http://linuxecke.volkoh.de/vokoscreen/vokoscreen-2.5.0.tar.gz Could you download it and give it a try?

If this does not help, you could also try to change the video container and/or the video codec settings.

**Comment by dtschmitz** [comment]
Many things depend on ffmpeg.  It would be difficult to uninstall without uninstalling a raft of other apps, which if at all possible I would prefer not to do.

If I may make a suggestion -- something for you to consider:  New technology like AppImage: http://appimage.org/

Since there is a myriad of different Distros, reduce complexity by folding your App into its own optimal environmnent with settings that will be guaranteed to work, regardless of which Distro the end-user has.  Yes?

For example: Openshot AppImage: http://www.openshot.org/download/

I think this is the time for you to give this a try.  I will be happy to be your tester should you decide to go in this direction.  For now, I am going to close this issue.  Best Wishes -- Dietrich

**Comment by dtschmitz** [comment]
BTW, I wrote this story several years ago.  Read and understand why I suggest AppImage:
http://linuxadvocates.blogspot.com/2013/03/the-linux-desktop-mess.html
Cheers!

**Comment by probonopd** [comment]
Please try the inofficial AppImage that I made for testing purposes:
https://bintray.com/probono/AppImages/vokoscreen/_latestVersion#files

It re-uses the official binaries:
https://github.com/AppImage/AppImages/blob/master/recipes/meta/vokoscreen.yml

**Comment by dtschmitz** [comment]
It still flickers with the AppImage Run version. :(
[vokoscreen-2017-06-11_12-52-04.zip](https://github.com/vkohaupt/vokoscreen/files/1066601/vokoscreen-2017-06-11_12-52-04.zip)
Sigh.  Hokay.  Let me know what I can do.  I am encouraged to see this AppImage!  Great Work!

**Comment by probonopd** [comment]
Sorry I have no idea why it flickers on your system. Can you, just for testing, run this on e.g., using a Ubuntu Live USB system? Just to exclude the possibility that it has to do with your OS setup.

**Comment by probonopd** [comment]
Just to narrow down the problem, you might also want to test another screenrecorder for comparison, https://bintray.com/probono/AppImages/SimpleScreenRecorder/_latestVersion#files. Does it work there?

**Comment by dtschmitz** [comment]
[vokoscreen-2017-06-11_14-02-52.mkv.tar.gz](https://github.com/vkohaupt/vokoscreen/files/1066650/vokoscreen-2017-06-11_14-02-52.mkv.tar.gz)
I installed Ubuntu-Gnome 17.04 into a qemu-kvm virtual machine and vokoscreen binary from the Ubuntu repository works fine per the above.
I don't know what this means or how it helps to troubleshoot and isolate the problem in my host environment, but, there you have it.

**Comment by dtschmitz** [comment]
Folks, I will close this ticket for now.  Thanks for trying.

**Comment by dtschmitz** [comment]
Ok, I am really confused, but I have good news.  I uninstalled vokoscreen and reinstalled:
[dts@ThinkPad ~]$ yaourt -Ss vokoscreen
aur/vokoscreen 2.5.4.beta-1 (77) (1.02)
    An easy to use screencast creator
aur/vokoscreen-git 2.5.3.beta.11.g7b2824f-2 [installed: 2.5.4.beta.24.gda7c449-1] (5) (0.01)

I have no idea why it now works, but, just wanted to let you know -- all is well now.  Thanks All.
Dietrich

**Comment by dtschmitz** [comment]
I feel like I am losing my mind.  :P Just when I thought everything was fine, the flicker is back.
I've tried simplescreenrecorder and the output (mkv) produces flicker as well.  So, it would appear that the core Qt Vokoscreen is fine and the modality of this issue is with Gnome-Shell and enable-animations and/or ffmpeg.
If I can be of assistance is cornering and solving this issue please let me know.

**Comment by tuxmuck** [comment]
Probiere halt einfach mal einen anderen Desktop.
Lässt sich doch einfach nachinstallieren ....

**Comment by dtschmitz** [comment]
Das ist keine lösung.

**Comment by J5lx** [comment]
Du must ja nicht komplett auf einen anderen Desktop umsteigen,sondern einfach nur vorübergehend einen installieren und anschmeißen um herauszufinden ob das Problem da auch auftritt.

**Comment by tuxmuck** [comment]
Aber wenigstens mal ein Test.
Wenn Gnome ein irgendwie geartetes Problem hat, könnte man es wenigstens eingrenzen.
Der nächste Schritt wäre dann, mal ältere ffmpeg Versionen zu probieren.

Hätte vokoscreen ein Problem, dann würden hier oder über den Support viel mehr Meldungen aufschlagen. Dem ist aber nicht so.

**Comment by dtschmitz** [comment]
Ich habe Plasma getestet. Beachten Sie, wie die Vokoscreen-Tasten flackern
[vokoscreen-2017-06-14_07-19-18.zip](https://github.com/vkohaupt/vokoscreen/files/1074509/vokoscreen-2017-06-14_07-19-18.zip)

**Comment by dtschmitz** [comment]
[vokoscreen-2017-06-14_08-16-39.zip](https://github.com/vkohaupt/vokoscreen/files/1074523/vokoscreen-2017-06-14_08-16-39.zip)
Das ist gnome-shell mit "enable-animationen" deaktiviert. Beachten Sie, wie die Ausgabe nicht die Fensterkontextänderung zeigt, wenn ich auf jedes Fenster klicke. Sehr eigenartig.

**Comment by dtschmitz** [comment]
Beim Versuch der AppImage gibt es kein Bildschirmflimmern, aber beim Umschalten von Windows-Kontext wird die Ausgabe nicht aktualisiert, um die Änderung des modalen Fensters anzuzeigen. Ganz eigenartig.
[vokoscreen-2017-06-14_08-32-30.zip](https://github.com/vkohaupt/vokoscreen/files/1074564/vokoscreen-2017-06-14_08-32-30.zip)

**Comment by tuxmuck** [comment]
Was mir in allen Aufnahmen auffällt - unten in der Statuszeile von vokoscreen wird bei frames und Dateigrösse immer nur 0 / 0 angezeigt. Hier müsste aber bei laufender Aufnahme dynamisch immer der jeweils aktuelle Wert angezeigt werden.

Ich vermute da einen granatenmässigen Murks irgendwo im System - wo/was kann ich nicht sagen.
Denn wie ich schon schrieb, auf einem Antergos hier läuft es einwandfrei.

**Comment by dtschmitz** [comment]
Es gibt definitiv gremlins in meinem system. Seufzer.

**Comment by dtschmitz** [comment]
I just checked gnome-logs and see this in the application log for Vokoscreen:
http://i.imgur.com/WsDRvIq.png
Failed to open VDPAU backend libvdpau_va_gl.so: cannot open shared object file: No such file or directory

**Comment by dtschmitz** [comment]
Looking at your voko log output: I see you make a call to ffmpeg with arguments.  I have dropped to a terminal prompt and run ffmpeg with your args.  It returns the following ouput:

/****************
[dts@ThinkPad vokoscreen]$ /usr/bin/ffmpeg -report -f x11grab -draw_mouse 1 -framerate 25 -video_size 1366x768 -i :0+0,0 -f pulse -i vokoscreenMix.monitor -pix_fmt yuv420p -c:v libx264 -preset veryfast -c:a libmp3lame -q:v 1 -s 1366x768 -f matroska /home/dts/Videos/.vokoscreen-recording/vokoscreen-test.log
ffmpeg started on 2017-06-14 at 09:48:17
Report written to "ffmpeg-20170614-094817.log"
ffmpeg version 3.3.2 Copyright (c) 2000-2017 the FFmpeg developers
  built with gcc 7.1.1 (GCC) 20170516
  configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-avisynth --enable-avresample --enable-fontconfig --enable-gmp --enable-gnutls --enable-gpl --enable-ladspa --enable-libass --enable-libbluray --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libiec61883 --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxvid --enable-shared --enable-version3
  libavutil      55. 58.100 / 55. 58.100
  libavcodec     57. 89.100 / 57. 89.100
  libavformat    57. 71.100 / 57. 71.100
  libavdevice    57.  6.100 / 57.  6.100
  libavfilter     6. 82.100 /  6. 82.100
  libavresample   3.  5.  0 /  3.  5.  0
  libswscale      4.  6.100 /  4.  6.100
  libswresample   2.  7.100 /  2.  7.100
  libpostproc    54.  5.100 / 54.  5.100
[x11grab @ 0x55fc662c72a0] Stream #0: not enough frames to estimate rate; consider increasing probesize
Input #0, x11grab, from ':0+0,0':
  Duration: N/A, start: 1497448097.221111, bitrate: N/A
    Stream #0:0: Video: rawvideo (BGR[0] / 0x524742), bgr0, 1366x768, 25 fps, 1000k tbr, 1000k tbn, 1000k tbc
vokoscreenMix.monitor: Input/output error
[dts@ThinkPad vokoscreen]$ 

***************/
Am I interpreting this correctly?  It doesn't run and raises a silent exception for x11grab.

**Comment by dtschmitz** [comment]
Looks like x11grab is not an error after all.  Will keep digging.  There's gold in them thar hills. ;)

**Comment by debuglevel** [comment]
@tuxmuck most virtual machines like VirtualBox use a rather basic virtual graphics adapter. If they support advanced features like 3D acceleration, you're lucky already. Therefore, it is not a big surprise, that it works in a VM. In fact, regarding the graphics adapter, vokoscreen would probably run on every distribution on every physical host as long as a VM is used (because it's just the same graphics adapter, the guest Linux uses).

The only bugs one could detect by using a VM, is whether the software side is scrambling up something (e.g. incompatible packages are installed, the desktop environment is messing with x11grab, etc). A VM is not suitable to detect the various problems with bad graphics drivers, testing various graphic adapters (you do not even test your own graphics adapter, but VirtualBox's emulated one) or even general problems with 3D acceleration and composite managers.

**Comment by debuglevel** [comment]
@dtschmitz does your Antergos also run inside KVM? If yes: you could install a fresh Antergos in a KVM and test if there is a general problem with the drivers or desktop environment, Antergos is shipping. If the bug does occur, you would at least know, whether the bug is not in your main system but a general bug with Antergos in combintion with your hardware.

**Comment by dtschmitz** [comment]
@debuglevel Yes Marc.  This is a core i5 with bare metal hypervisor support, so I can set up a test for you of a kvm instance of Antergos to see if vokoscreen runs from a clean install.  That makes eminent sense.  Will do and report back with results.  Cheers!

**Comment by dtschmitz** [comment]
So, okay, something is conflicting in my machine's installation that is influencing this issue.
A kvm clean install of Antergos Gnome-Shell and tested with the below output.  Vokoscreen works fine.
[vokoscreen-2017-06-15_12-43-01.zip](https://github.com/vkohaupt/vokoscreen/files/1077628/vokoscreen-2017-06-15_12-43-01.zip)
I don't know if this is solvable as it appears 'the mix' of installables on my local machine are contributing to the problem.  Which means there's nothing wrong with Vokoscreen.

**Comment by dtschmitz** [comment]
Just an fyi.  I just did a clean install of Antergos minimal Gnome Shell -- went straight to installing Vokoscreen -- it's still flickering.  (sobs... Mommy)  I don't expect you to resolve this.
[vokoscreen-2017-06-17_00-29-51.zip](https://github.com/vkohaupt/vokoscreen/files/1082216/vokoscreen-2017-06-17_00-29-51.zip)

**Comment by vkohaupt** [comment]
Hello,
i am return from Holiday and i see many postings.
Please start vokoscreen from a terminal and start a record, then click 
on every TAB, then stop the record.

Then copy the full terminal output and show this here and the send the 
video.

**Comment by dtschmitz** [comment]
[vokoscreen-2017-06-17_07-46-43.zip](https://github.com/vkohaupt/vokoscreen/files/1082520/vokoscreen-2017-06-17_07-46-43.zip)

**Comment by vkohaupt** [comment]
Once more.

I miss the terminal output.

And bevor you record a video please activate Showclick, i want see every 
mouseclick.

**Comment by dtschmitz** [comment]
[terminal-output.zip](https://github.com/vkohaupt/vokoscreen/files/1082534/terminal-output.zip)
[vokoscreen-2017-06-17_08-22-04.zip](https://github.com/vkohaupt/vokoscreen/files/1082535/vokoscreen-2017-06-17_08-22-04.zip)

**Comment by vkohaupt** [comment]
I have a tested solution for Antergos(Gnome) and KDE(Plasma5) for Intel Graphic cards
I see on my System no flickering in a Video
Please extract the ZIP in your home, then log off or restart your system.

[drirc.zip](https://github.com/vkohaupt/vokoscreen/files/1082809/drirc.zip)

**Comment by dtschmitz** [comment]
[terminal-output2.zip](https://github.com/vkohaupt/vokoscreen/files/1082818/terminal-output2.zip)
[vokoscreen-2017-06-17_19-34-37.zip](https://github.com/vkohaupt/vokoscreen/files/1082819/vokoscreen-2017-06-17_19-34-37.zip)
No screen flicker, but odd that screen clicks isn't 'where' it should be (instead lower left corner of Voko dialog -- and clicking the tabs doesn't seem to be getting an event, yet I see the tab changes just fine.

Twilight Zone. :)

**Comment by dtschmitz** [comment]
You are welcome to come on my system using chrome-remote and see for yourself.

**Comment by dtschmitz** [comment]
Just an fyi, I don't know if it matters but the default ui for Gnome-Shell 3.24.2 is Wayland.

**Comment by vkohaupt** [comment]
[dts@ThinkPad vokoscreen]$ vokoscreen &
[1] 6822
[dts@ThinkPad vokoscreen]$ qt5ct: using qt5ct plugin

Is this your Twilight Zone?

**Comment by dtschmitz** [comment]
So, should I install the Qt4 plugin?
BTW, I've been testing the 'out of the box' install of Antergos -- which uses lightdm (vs. gdm)
Regardless of 'which' display driver I choose, it runs with X.org. :/
I've disabled lightdm and installed gdm and it's running Xwayland. :)
So, install Qt4 plugin, yes? and try again?

**Comment by dtschmitz** [comment]
[vokoscreen-2017-06-18_09-47-02.zip](https://github.com/vkohaupt/vokoscreen/files/1083141/vokoscreen-2017-06-18_09-47-02.zip)
I have intalled Qt4, installed your xml zip, and when running from an Xwayland display driver, do not get screen flicker -- instead the entire screen is blank but it shows the cursor movement.

Just poke my eyes out with a sharpened #2 pencil. ;)

**Comment by vkohaupt** [comment]
> I have intalled Qt4, installed your xml zip, and when running from an Xwayland display driver, do not get screen flicker -- instead the entire screen is blank but it shows the cursor movement.

At the moment is the focus not XWayland, Wayland, ... and can not say 
anything about it.


> Just poke my eyes out with a sharpened #2 pencil. ;)

i´m not understand, is this an american joke?

**Comment by dtschmitz** [comment]
It was my attempt at humor.  Yes, it's an American joke.  You had to be there.
Anyhow, I will revert to X.org if that is where you wish to continue this matter.
Just curious, does Vokoscreen support XWayland?:
[dts@ThinkPad ~]$ ps -ef | grep Xwaygdm        493   462  0 13:36 tty1     00:00:00 /usr/bin/Xwayland :1024 -rootless -noreset -listen 4 -listen 5 -displayfd 6
dts        771   746  0 13:36 tty2     00:00:33 /usr/bin/Xwayland :0 -rootless -noreset -listen 4 -listen 5 -displayfd 6

Let me know how to proceed.  Thank you!

**Comment by dtschmitz** [comment]
Bump

**Comment by vkohaupt** [comment]
Dietrich T. Schmitz schrieb am 21.06.2017 um 14:22:
> Bump
> 
https://www.youtube.com/watch?v=DkhrmvtnX_I :>)

**Comment by dtschmitz** [comment]
Hah.  Double Bump.  Seriously, What's up?

**Comment by vkohaupt** [comment]
Dietrich T. Schmitz schrieb am 21.06.2017 um 20:01:
> Hah.  Double Bump.  Seriously, What's up?

My first thought was Lady Bump from Penny McLean.

What you mean with Bump?

**Comment by dtschmitz** [comment]
This is in our terminal output when vlc is called to 'Play' the video:
/*
[vokoscreen] ---Begin Pulse unload Module--- 
[vokoscreen] pactl unload-module number "30" 
[vokoscreen] pactl unload-module number "29" 
[vokoscreen] pactl unload-module number "28" 
[vokoscreen] ---End Pulse unload Module--- 
  
[vokoscreen] play video:  "/usr/bin/vlc \"/home/dts/Videos/vokoscreen-2017-06-21_15-10-20.mkv\"" 
"[0000000001f5b148] core libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.\nlibva info: VA-API version 0.40.0\nlibva info: va_getDriverName() returns 0\nlibva info: Trying to open /usr/lib/dri/i965_drv_video.so\nlibva info: Found init function __vaDriverInit_0_40\nlibva info: va_openDriver() returns 0\n[00007f97940009c8] vdpau_avcodec generic error: decoder profile not supported: 12\nlibva info: VA-API version 0.40.0\nlibva info: va_getDriverName() returns 0\nlibva info: Trying to open /usr/lib/dri/i965_drv_video.so\nlibva info: Found init function __vaDriverInit_0_40\nlibva info: va_openDriver() returns 0\n" 
[vokoscreen] ---Begin Pulse unload Module--- 
[vokoscreen] ---End Pulse unload Module--- 

What is your interpretation of this?

**Comment by dtschmitz** [comment]
Used to 'nudge' someone to take action on an issue/comment that has not been responded to. ;)

**Comment by vkohaupt** [comment]
Dietrich T. Schmitz schrieb am 21.06.2017 um 21:23:
> This is in our terminal output when vlc is called to 'Play' the video:
> /*
> [vokoscreen] ---Begin Pulse unload Module---
> [vokoscreen] pactl unload-module number "30"
> [vokoscreen] pactl unload-module number "29"
> [vokoscreen] pactl unload-module number "28"
> [vokoscreen] ---End Pulse unload Module---

This comes by end of recording and say me you have record with two Audio 
devices. If record finish we must unload in reverse sequence.


> "[0000000001f5b148] core libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.\nlibva info: VA-API version 0.40.0\nlibva info: va_getDriverName() returns 0\nlibva info: Trying to open /usr/lib/dri/i965_drv_video.so\nlibva info: Found init function __vaDriverInit_0_40\nlibva info: va_openDriver() returns 0\n[00007f97940009c8] vdpau_avcodec generic error: decoder profile not supported: 12\nlibva info: VA-API version 0.40.0\nlibva info: va_getDriverName() returns 0\nlibva info: Trying to open /usr/lib/dri/i965_drv_video.so\nlibva info: Found init function __vaDriverInit_0_40\nlibva info: va_openDriver() returns 0\n"

This is an info from VLC

By my system vls used hardware decoding
[..]
vcodec decoder: Using Intel i965 driver for Intel(R) Ivybridge Desktop - 
1.7.0 for hardware decoding.

**Comment by vkohaupt** [comment]
> Just curious, does Vokoscreen support XWayland?:

Vokoscreen is a X11(xcb) Application and (your) system support XWayland, 
so the vokoscreen GUI can show on a Wayland-Desktop.

Vokoscreen support not Wayland, that we can see, if we have a Wayland 
Application on the screen, the tools showklick and showkey do not. Is a 
X11 Application on the screen this two tools do.

And a record comes with a black screen and only show the mouse.


> [dts@ThinkPad ~]$ ps -ef | grep Xwaygdm        493   462  0 13:36 tty1     00:00:00 /usr/bin/Xwayland :1024 -rootless -noreset -listen 4 -listen 5 -displayfd 6
> dts        771   746  0 13:36 tty2     00:00:33 /usr/bin/Xwayland :0 -rootless -noreset -listen 4 -listen 5 -displayfd 6

My system opensuse leap 42.2 doesn't support Wayland, i can only test 
with Fedora 25 from a USB-Stick and i see thousand bugs.

**Comment by dtschmitz** [comment]
Yes, I agree.  I've tested it in Xwayland and the output in vlc is 'black'.
I've downloaded your vokoscreen.tar.gz and run it against doxygen so I can study it.
Okay, now I understand your pc is handicapped to test Wayland.  Hmmmm.
I used to use OpenSuSE (until Novell got in bed with Microsoft).
Then I spent a few years thinking Ubuntu was great.  No more.
Now, Antergos (Arch derivative) and I have not changed from it in 2 years and am not looking back. ;)
It's a 'no brainer' for me since it is rolling release.
But that is a 'double edge' sword and occasionally updates can go fubar.
To be proactive and take defensive measures, I have an external eSATA docking station with a 1TB HDD running in daily midnight cron Backintime which runs flawlessly like a swiss watch.
Also, I waited 1 year for Antergos to shake out the bugs in their ZFS implementation.  Now I have switched from ext4 to ZFS and have installed zfs-auto-snapshot which runs hourly, daily, weekly, monthly in systemd timers -- this allows instantaneous recovery from any fubar.  I can also mount any snapshot for selective file-by-file recovery.  A major improvement.

Anyhow, I am willing to help you in whatever way I can to address this issue.

I strongly urge you to try Antergos although I understand the history behind SuSe which would lead you to prefer it in Germany.  But it leaves you at a major disadvantage insofar as 'static' package management is concerned.

Arch and derivatives is pretty popular with Devs here in the USA.

Halrighty then.  Cheers!

**Comment by dtschmitz** [comment]
BTW, I found 'Green Recorder' and tested it.  It works with Xwayland.
You might take a look at the source code (Python) and see if you can glean what is being done there if it helps you to refactor Xwayland support.

**Comment by dtschmitz** [comment]
Ran across this, fyi:
https://wayland.freedesktop.org/qt5.html

**Comment by dtschmitz** [comment]
A slight digression:  To the extent that I know you use openSUSE, have you looked at Tumbleweed?:
https://en.opensuse.org/Portal:Tumbleweed

**Comment by dtschmitz** [comment]
Greetings.  Are you working towards a resolution to this issue?  No word from you in 5 days.  Let me know, one way or the other.
Bitte Danke.

**Comment by vkohaupt** [comment]
Dietrich T. Schmitz schrieb am 02.07.2017 um 23:54:
> Greetings.  Are you working towards a resolution to this issue?

To time work on a other project, a must to earn money.

Vokoscreen is a hobby project and i can not life from this.

**Comment by dtschmitz** [comment]
Well, okay.  I understand.  Closing issue.

**Comment by dtschmitz** [comment]
I know you've missed me so I thought I'd stop by and say hi.  Hi!
Other than that, I have found since last posting above that upon clean install of Antergos GNOME, that Vokoscreen works fine.  Noting that GNOME in Antergos uses the lightdm display manager.  The other day, I chose to switch to Manjaro GNOME which uses gdm and I am now back to the same issue as above -- It now makes me think there is a lightdm modality in which Vokoscreen works fine.  I just wish it could be narrowed down to 'what' may be causing the problem.  

There.  I feel better.  Got that off my chest. ;)
BTW, I installed lightdm but it errors out in systemd when I try to start it.  This is an open bug on Manjaro so I will wait for a fix and try lightdm when fixed.

Kind Regards Dietrich
